### ISSUE TITLE
Unable to order contents by step if you have a [NotAssigned ] content in the list 

### ISSUE THREAD

**Issue by bryanboza** [dotCMS/core#14897]
We need group the [NotAssigned] contents 

## Current Behavior
When you reset the workflow of some piece of content, this leaves the content in a [NotAssigned] step and if you try to order the list by step this isn't working 

## Expected Behavior
We need to group the [NotAssigned] and order alphabetically 
SC: https://screencast.com/t/axwOkl68ci

## Your Environment
Tested on master // Postgres // FF

**Comment by jgambarios** [comment]
PR: https://github.com/dotCMS/core/pull/14898

**Comment by jgambarios** [comment]
**Note to QA:** For now as the value `[Not Assigned]` is calculated the `Step` column won't be sortable.

**Comment by jgambarios** [comment]
We removed the ability to sort by the `Step` column because it is a calculated value as a temporal fix, for 5.0.1 we need to be able to sort by that column but that means we need to add the current step (or something similar) into the contentlet index in order to apply properly the sort.

This change needs to be in two places as in two places we disable the sorting, Content Search (Described on this ticket PR: https://github.com/dotCMS/core/pull/14898) and when relating content in the edit mode https://github.com/dotCMS/core/issues/14939 (PR: https://github.com/dotCMS/core/pull/14958)

**Comment by john-thomas-dotcms** [comment]
+1 to fix this.  This one is pretty important to have with the new workflow changes, since it's difficult to clean up the workflow steps and make sure all content is in a proper step without being to sort the content by step.

It seems like it would make sense to add the current step into the contentlet index, as I'm guessing it would help to have that in a number of places where content is accessed.  The more we transition to using workflows (as opposed to "default" actions of various types) on workflows, the more useful it's likely to be.

**Comment by dsilvam** [comment]
PR: https://github.com/dotCMS/core/pull/15123

**Comment by jdotcms** [comment]
Unable to reproduce

**Comment by bryanboza** [comment]
After retesting the card, now we have problems trying to use this column to order in the content dialog when you are adding a new content to any page, and you try to order the list of contents by step, we are unable to order and show a blank list 
SC: https://screencast.com/t/ceLl23pLeTR
Also, we have this error on log
Log: https://gist.github.com/bryanboza/e4d1f36c0237a99809ebc75efc1f2cb5

**Comment by bryanboza** [comment]
Fixed, now we are able to order by step without problems

**Comment by dsilvam** [comment]
Sorting by Step not working from the Content search dialog when adding content to a page. Emtpy list after clicking on the step column and following error in the log:
```
Caused by: org.elasticsearch.index.query.QueryShardException: No mapping found for [webpagecontent.__wfstep___dotraw] in order to sort on
    at org.elasticsearch.search.sort.FieldSortBuilder.build(FieldSortBuilder.java:319) ~[elasticsearch-6.1.3.jar:6.1.3]
```

**Comment by jgambarios** [comment]
PR: https://github.com/dotCMS/core/pull/15316

**Comment by bryanboza** [comment]
Fixed, tested on release-5.0.2 and now I'm unable to reproduce the problem
